odl/contrib/spdhg/example_spdhg_PET_1k.py

```diff
@@ -0,0 +1,681 @@
+# ----------------------------------------------------------------------
+# Matthias J. Ehrhardt
+# Cambridge Image Analysis, University of Cambridge, UK
+# m.j.ehrhardt@damtp.cam.ac.uk
+#
+# Copyright 2017 University of Cambridge
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# ----------------------------------------------------------------------
```

Comment:
Just a comment upfront, before someone starts reviewing. 

In the interest of avoiding infinite hassle, we *can't have mixed licenses* in one package, even if it's under `contrib`. The stuff is shipped with ODL either way, and anybody who relies on the code being MPL couldn't use it anymore without worries if they would always have to look through all files and see if there's a different license lurking somewhere.

It's also written in [the guidelines](https://github.com/odlgroup/odl/tree/master/odl/contrib#scope).

So if you need to keep the Apache License, that's perfectly fine, but then the code needs to be in its own repository.